### PR TITLE
Allow sword of beheading drop piglin heads

### PR DESCRIPTION
**Pull request by ybw0014**
## Description

According to Minecraft wiki [Head](https://minecraft.fandom.com/wiki/Head) page, piglin head is available but there is no way to get it through Sword of Beheading. I think it should drop a piglin head when killing a piglin.

## Proposed changes

- Use enhanced switch.
- Add piglin head as drop when killing piglins.

## Related Issues (if applicable)

N/A

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/SwordOfBeheading.java [hunk 1]
```diff
@@ -86,9 +94,9 @@ public EntityKillHandler getItemHandler() {
 
                         e.getDrops().add(skull);
                     }
-                    break;
-                default:
-                    break;
+                }
+                default -> {
+                }
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/SwordOfBeheading.java:99]: I believe you can remove this entirely right? Or just inline it if not

**Review by JustAHuman-xD — COMMENTED** [review]
May be out of scope for the PR but I wonder if theirs a better way to do that, currently it's very copy paste in the switch statement

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/SwordOfBeheading.java (hunk 1 above)
Comment (ybw0014) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/SwordOfBeheading.java]: biscuit wanted me to keep default in one of my previous PR, so i'll just inline it.

**Review by J3fftw1 — DISMISSED** [review]
i hate this code but I will approve it since this is how it has been done before.

**Review by JustAHuman-xD — APPROVED** [review]
LGTM and I agree with Jeff xd

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/SwordOfBeheading.java [hunk 2]
```diff
@@ -56,27 +59,32 @@ public EntityKillHandler getItemHandler() {
...
+                case WITHER_SKELETON -> {
                     if (random.nextInt(100) < chanceWitherSkeleton.getValue()) {
                         e.getDrops().add(new ItemStack(Material.WITHER_SKELETON_SKULL));
                     }
-                    break;
-                case PLAYER:
+                }
+                case PIGLIN -> {
+                    if (random.nextInt(100) < chancePiglin.getValue()) {
+                        e.getDrops().add(new ItemStack(Material.PIGLIN_HEAD));
```
Comment (Sefiraat) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/SwordOfBeheading.java:84]: Do we need a version check here, looks like this skull was added in 1.20? (ignoring the experimental pack one)

Comment (JustAHuman-xD) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/weapons/SwordOfBeheading.java:84]: I was wondering about that but in the checklist they said they tested it across versions so I assumed it failed silently, good question tho

**Review by Sfiguz7 — CHANGES_REQUESTED** [review]
Marking requested changes until the conversation above is resolved just in case it turns out to be something important. Feel free to ping me for reapproval if it is confirmed to be ininfluential

**Review by JustAHuman-xD — APPROVED** [review]
LGTM, tested on 1.20.1 and all heads (apart from player cause I had no other players) dropped as they should

**Review by JustAHuman-xD — APPROVED** [review]
LGTM

**Review by J3fftw1 — APPROVED** [review]
I dont like how this is done but LGTM

**Review by Sfiguz7 — APPROVED** [review]
Going off of my previous comment, LGTM but want a double check

**Review by Sfiguz7 — CHANGES_REQUESTED** [review]
Actually need to make it a request changes just in case, @Slimefun/bug-testers as soon as you have news please ping me on Discord

**Review by Sfiguz7 — APPROVED** [review]
I swear I am not high, just saw it now has the version check. I need to shut up

**Pull request merged**